sourmash_lib/lca/command_classify.py

```diff
@@ -0,0 +1,249 @@
+        if count < threshold:
+            break
+
+        n += 1
+
+        # construct [(rank, name), ...] lineage
+        lineage = []
+        parent = lca
+        while parent:
+            lineage.insert(0, parent)
+            tree_counts[parent] += count
+            parent = parents.get(parent)
+        debug(n, count, lineage[1:])
+
+        # update tree with this set of assignments
+        lca_utils.build_tree([lineage], tree)
+
+    if n > 1:
+        debug('XXX', n)
+
+    status = 'nomatch'
+    if not tree:
+        return [('', '')], status
+
+    # now find least-common-ancestor of the resulting tree.
+    lca, reason = lca_utils.find_lca(tree)
+    if reason == 0:               # leaf node
+        status = 'found'
+        debug('END', lca)
+    else:                         # internal node => disagreement
+        status = 'disagree'
+        debug('MULTI', lca)
+
+    # backtrack to full lineage via parents
+    lineage = []
+    parent = lca
+    while parent != ('root', 'root'):
+        lineage.insert(0, parent)
+        parent = parents.get(parent)
+
+    debug(parents)
+    debug('lineage is:', lineage)
+
+    return lineage, status
+
+
+def classify(args):
+    """
+    main single-genome classification function: 
+    """
+    p = argparse.ArgumentParser()
+    p.add_argument('--db', nargs='+', action='append')
+    p.add_argument('--query', nargs='+', action='append')
+    p.add_argument('--threshold', type=int, default=DEFAULT_THRESHOLD)
+    p.add_argument('-o', '--output', type=argparse.FileType('wt'),
+                   help='output CSV to this file instead of stdout')
+    p.add_argument('--traverse-directory', action='store_true',
+                        help='load all signatures underneath directories.')
+    p.add_argument('-d', '--debug', action='store_true')
+    args = p.parse_args(args)
+
+    if not args.db:
+        error('Error! must specify at least one LCA database with --db')
+        sys.exit(-1)
+
+    if not args.query:
+        error('Error! must specify at least one query signature with --query')
+        sys.exit(-1)
+
+    if args.debug:
+        set_debug(args.debug)
+
+    ksize_vals = set()
+    scaled_vals = set()
+    dblist = []
+
+    # flatten --db and --query
+    args.db = [item for sublist in args.db for item in sublist]
+    args.query = [item for sublist in args.query for item in sublist]
+
+    # load all the databases
+    for db_name in args.db:
+        notify(u'\r\033[K', end=u'', file=sys.stderr)
+        notify('... loading database {}'.format(db_name), end='\r',
+              file=sys.stderr)
+
+        lca_db = lca_utils.LCA_Database()
+        lca_db.load(db_name)
+
+        ksize_vals.add(lca_db.ksize)
+        if len(ksize_vals) > 1:
+            raise Exception('multiple ksizes, quitting')
+        scaled_vals.add(lca_db.scaled)
+        if len(scaled_vals) > 1:
+            raise Exception('multiple scaled vals, quitting')
```

Comment:
Maybe downsample the largest one to the smaller one, instead of failing?

Reply:
(maybe this is more appropriate for signatures, but we could also do it for databases)

Reply:
yes, it would be easy to do here. (for SBTs we have a --scaled parameter on searches already.)